### PR TITLE
DFAST-95: Refactor estimate_sedimentation_length method

### PR DESCRIPTION
**Pull request by MRVermeulenDeltares**

### CODE REVIEW

File: dfastmi/kernel/core.py [hunk 1]
```diff
@@ -244,10 +212,9 @@ def estimate_sedimentation_length2(
     L : float
         The expected yearly impacted sedimentation length [m].
     """
-    Lt = [tmi[i] * celerity[i] for i in range(len(tmi))]
+    yearly_sedimentation_lengths = [tmi[i] * celerity[i] for i in range(len(tmi))]
```
Comment (hrajagers) [inline dfastmi/kernel/core.py:215]: The term `yearly_sedimentation_lengths` isn't appropriate. The variable represents the contribution of each period/discharge level to the total sedimentation length, so let's use `sedimentation_length_contributions` instead.

File: dfastmi/kernel/core.py [hunk 2]
```diff
@@ -244,10 +212,9 @@ def estimate_sedimentation_length2(
...
     
-    return sum(Lt) * 1000
-
+    return sum(yearly_sedimentation_lengths) * 1000
```
Comment (hrajagers) [inline dfastmi/kernel/core.py:217]: Let's introduce a constant `km_to_m` equal to 1000 to show where it's coming from.

**Review by hrajagers — CHANGES_REQUESTED** [review]
The moving of code around looks fine. Just two small comments about the improvements to `core.estimate_sedimentation_length`.

**Comment by hrajagers** [comment]
Hi @MRVermeulenDeltares, I removed the call to kernel.legacy.sedimentation_length ... so, you can remove that legacy function and the associated tests.

I've left my recommendations for the newer kernel.core.sedimentation_length to you.

File: dfastmi/kernel/core.py (hunk 1 above)
Comment (MRVermeulenDeltares) [reply dfastmi/kernel/core.py:215]: I will change this.

File: dfastmi/kernel/core.py (hunk 2 above)
Comment (MRVermeulenDeltares) [reply dfastmi/kernel/core.py:217]: I will add this, I will use capitals, since constants should be capital in python I think.

**Comment by ralph-peelen-deltares** [comment]
As discussed, the sonar cloud issues are not related to the resolution

**Pull request merged**